app/service/collection.py

```diff
@@ -109,24 +107,26 @@ def validate_multiple_ids(
 
 
 @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
-def validate(import_ids: set[str], db: Session = db_session.get_db()) -> None:
+def validate(import_ids: set[str], db: Optional[Session]) -> None:
```

Comment:
This is going to make Bandit so so so happy - the db call in the function signature was a vulnerability!